status/views.py

```diff
@@ -0,0 +1,107 @@
+"""
+Status checks for LORE.
+
+Notes:
+
+* Useful messages are logged, but "no config found" is returned whether
+  settings are missing or invalid, to prevent information leakage.
+* Different services provide different information, but all should return
+  "up," "down", or "no config found" for the "status" key.
+"""
+
+import re
+from datetime import datetime
+import logging
+
+from django.conf import settings
+from django.http import JsonResponse
+from elasticsearch import Elasticsearch, ConnectionError as ESConnectionError
+from psycopg2 import connect, OperationalError
+from redis import StrictRedis, ConnectionError as RedisConnectionError
+
+log = logging.getLogger(__name__)
+
+
+def get_pg_info():
+    """Check PostgreSQL connection."""
+    try:
+        conf = settings.DATABASES['default']
+    except (AttributeError, KeyError):
+        log.error("No PostgreSQL connection info found in settings.")
+        return {"status": "no config found"}
+    try:
+        start = datetime.now()
+        connect(
+            database=conf["NAME"], user=conf["USER"], host=conf["HOST"],
+            port=conf["PORT"], password=conf["PASSWORD"]
+        )
+        micro = (datetime.now() - start).microseconds
+    except (OperationalError, KeyError):
+        log.error("Invalid PostgreSQL connection info in settings: %s", conf)
+        return {"status": "down"}
+    return {"status": "up", "response_ms": micro}
+
+
+def get_redis_info():
+    """Check Redis connection."""
+    pat = re.compile(r'redis://([^:]+):(\d+)/(\d+)$')
+    try:
+        url = settings.BROKER_URL
+    except AttributeError:
```

Comment:
To be more specific: https://github.com/celery/celery/blob/a7c168d6df82bcd060e02e647a916d61627959dc/celery/backends/redis.py#L59 and use connparams from that instantiated class to do the connection or just use its client property.